save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/organizations/OrganizationSettingsMenu.kt

```diff
@@ -150,14 +145,17 @@ private fun organizationSettingsMenu() = FC<OrganizationSettingsMenuProps> { pro
                 }
                 div {
                     className = ClassName("d-sm-flex align-items-center justify-content-center p-3")
-                    button {
-                        type = ButtonType.button
-                        className = ClassName("btn btn-sm btn-danger")
-                        disabled = !props.selfRole.hasDeletePermission()
-                        onClick = {
-                            props.deleteOrganizationCallback()
+                    deleteOrganizationFun {
+                        organizationName = props.organizationName
+                        onDeletionSuccess = {
+                            window.location.href = "${window.location.origin}/"
+                        }
+                        buttonStyleBuilder = { childrenBuilder ->
+                            with(childrenBuilder) {
```

Comment:
why do we need here `with`?

Reply:
Otherwise, it is impossible to transfer the picture

Reply:
Got it. This field expects receiver as parameter. Can we refactor it?